transport/http/outbound.go

```diff
@@ -282,7 +282,7 @@ func (o *Outbound) callWithPeer(
 		// maintenance loop resumes probing for availability.
 		p.OnDisconnected()
 
-		return nil, err
+		return nil, yarpcerrors.UnknownErrorf("unknown error from http client: %s", err.Error())
```

Comment:
uuuuugh, go errors are annoying, looking at the http internals, there are a bunch of errors that can be returned from the client.Do, some of which are connection failures, but we don't seem to have strict guarantees on that.  There are configuration, EOF, timeout.  And unfortunately, a lot of these errors don't have types we can assert against...

This is a small subset of the errors: https://github.com/golang/go/blob/master/src/net/http/transport.go#L641-L668

We probably can't validate that these are "unavailable" since there are a lot of error types returned from the client.Do function...